Lib/collections/__init__.py

```diff
@@ -1032,7 +1032,7 @@ def fromkeys(cls, iterable, *args):
 
     def copy(self):
         'New ChainMap or subclass with a new copy of maps[0] and refs to maps[1:]'
-        return self.__class__(self.maps[0].copy(), *self.maps[1:])
+        return self.__class__(self.maps[0].copy(), *_islice(self.maps, 1, None))
```

Comment:
This change doesn't do anything useful.  The time saved by not making a list is offset by the time to create the `islice` object.

On the receiving end, the `__init__()` method converts the input to a list.  This goes faster if the input is already a list than for an iterator . The former has a fast path and can take advantage of a length hint.